v0/destinations/hs/util.js

```diff
@@ -17,7 +136,187 @@ const getEmailAndUpdatedProps = properties => {
   };
 };
 
+/* NEW API util functions */
+
+/**
+ * look for the contact in hubspot and extract its contactId for updation
+ * @param {*} destination
+ * @returns
+ */
+const searchContacts = async (message, destination, lookupField = null) => {
+  const { Config } = destination;
+  let res;
+  let contactId;
+  const traits = getFieldValueFromMessage(message, "traits");
+  let propertyName;
+
+  // if propertyName (key name) is directly provided in this function
+  // eg: email
+  if (lookupField) {
+    propertyName = lookupField;
+  } else {
+    // look for propertyName (key name) in traits
+    // Config.lookupField -> lookupField
+    // traits: { lookupField: email }
+    propertyName = traits[`${Config.lookupField}`];
+  }
+
+  if (!propertyName) {
```

Comment:
Move this error throw outside of the function.

Reply:
the structure is changed now and all is being handled inside it.